### ISSUE TITLE
Very slow injection after update

### ISSUE THREAD

**Issue by nova-ind** [BrowserSync/browser-sync#695]
I'm using BS with gulp on a Windows machine. Until recently I had version 2.2.3, it worked great, the css was injected immediately after build. Since I updated to 2.7.12, it's been taking way longer to inject the CSS, like 5+ seconds after the build task finishes. I don't know why, haven't changed anything else in the gulpfile or on my local server (ports, IPs).

Any suggestions? I love BS, it's been a great time saver, but with this huge delay it's faster to manually reload the page. Let me know if I can do anything to help debug.

Cheers!

**Comment by shinnn** [comment]
https://github.com/BrowserSync/browser-sync/blob/f8c6a0ae33a1150a3cc7c4844a9dc1f8a6ba7407/CONTRIBUTING.md#-tips-for-a-creating-a-great-issue-report

**Comment by nova-ind** [comment]
Sorry about that, here's a more detailed report:

Win 7 x64
Gulp + BS in snippet mode
WAMP Server
Using it in a WordPress theme

The Sass and watch tasks:
```
// Compile Sass
gulp.task('sass', function() {
	gulp.src(path.styles.files)
		.pipe( gulpif(isDev, sourcemaps.init()) )
		.pipe( gulpif(isDev, sass(), sass({ outputStyle: 'compressed' })) )
		.on( 'error', function(err) { displayError(err); } )
		.pipe( concat('main.css') )
		.pipe( gulpif(isDev, sourcemaps.write(), autoprefixer('last 2 versions', 'ie 8')) )
		.pipe( gulp.dest(path.dest) )
		.pipe( browserSync.reload({stream:true}) );
});

// Compile Admin Sass
gulp.task('sass-admin', function() {
	gulp.src([path.styles.admin, path.styles.pluginAdmin])
		.pipe( gulpif(isDev, sourcemaps.init()) )
		.pipe( gulpif(isDev, sass(), sass({ outputStyle: 'compressed' })) )
		.on( 'error', function(err) { displayError(err); } )
		.pipe( concat('admin.css') )
		.pipe( gulpif(isDev, sourcemaps.write(), autoprefixer('last 2 versions', 'ie 8')) )
		.pipe( gulp.dest(path.dest) )
		.pipe( browserSync.reload({stream:true}) );
});

// Watch Sass files
gulp.task('watch', function() {
	browserSync.init({
		host: '192.168.0.103'
	});
	gulp.watch([path.styles.files], ['sass']);
	gulp.watch([path.styles.admin, path.styles.pluginAdmin], ['sass-admin']);
});
```

**Comment by shakyShane** [comment]
@nova-ind could you try this recipe and let us know if this is slow also. https://github.com/BrowserSync/recipes/tree/master/recipes/gulp.sass

**Comment by shinnn** [comment]
You should `return` gulp.src stream inside gulp.task. https://github.com/gulpjs/gulp/blob/86938061512f7633567026480dca03903ac07485/docs/API.md#return-a-stream

**Comment by nova-ind** [comment]
@shakyShane The recipe works well, the task takes about 8ms to finish and the changes are instant.
@shinnn Ah, I'm still a newb at this, I removed the return declarations while debugging an issue, and never put them back. The problem is still there, only instead of the task completing in ~8ms and BS detecting file change in ~5s, the whole task now takes ~5s.

I'm dumbfounded, it's so frustrating... I'll fiddle with the gulpfile some more, trim the tasks down, and report back if I find any solution. If you guys can suggest anything else I could try, it would be much appreciated, I'm running out of ideas.

**Comment by shakyShane** [comment]
@nova-ind we could only help further if provide a sample project highlighting the problem :)

**Comment by shinnn** [comment]
@shakyShane AFAIK `bs.reload({stream: true})` is deprecated and we should use [`bs.stream`](http://www.browsersync.io/docs/api/#api-stream) instead, right?

**Comment by shakyShane** [comment]
@shinnn yes that's true, although I doubt that has anything to do with this speed issue.

**Comment by shinnn** [comment]
> yes that's true

So, we should update the recipe. https://github.com/BrowserSync/recipes/tree/master/recipes/gulp.sass

**Comment by shakyShane** [comment]
Yep

**Comment by shinnn** [comment]
@nova-ind Can you show us the result of these commands?

```
$ node -v
$ npm -v
```

**Comment by nova-ind** [comment]
node v0.10.32
npm 2.1.18

I replaced bs.reload() with bs.stream(), there was indeed no improvement.

**Comment by shinnn** [comment]
You're using ancient Node and npm. Regardless of this issue I recommend using the latest io.js or Node, and npm.

**Comment by shinnn** [comment]
> there was indeed no improvement

https://github.com/BrowserSync/browser-sync/issues/695#issuecomment-115389758

**Comment by nova-ind** [comment]
Figured out why it is taking so long to complete the task, it's because I'm using sass functions to create color variants for different elements, with advanced color luminosity detection. The fact that I updated BS was just a coincidence. I had no idea sass functions could impact build time so much. If I replace the custom function with sass' built-in lightness(), it shaves 4s off, and that is for only 6 variants so far :(

I sincerely apologize for wasting your time, I hope I didn't cause too many headaches...
